clients/drcachesim/tracer/func_trace.cpp

```diff
@@ -101,7 +101,13 @@ static func_metadata_t *
 create_func_metadata(const char *name, int id, int arg_num, bool noret)
 {
     func_metadata_t *f = (func_metadata_t *)dr_global_alloc(sizeof(func_metadata_t));
-    strncpy(f->name, name, BUFFER_SIZE_ELEMENTS(f->name));
```

Comment:
This is a false positive then, not a true positive like in the PR description, since the `size` argument of `strncpy` is related to the destination buffer, not the source.

But then how would the suggested:
```
strncpy(f->name, name, BUFFER_SIZE_ELEMENTS(f->name) - 1);
NULL_TERMINATE_BUFFER(f->name);
```
solve your problem?
You don't see the error with `BUFFER_SIZE_ELEMENTS(f->name) - 1` ?

Does giving a name to the `size` argument of `strncpy` solve your problem (as you noticed in #7269)?